src/Microsoft.TestPlatform.ObjectModel/ConnectionInfo/TestRunnerConnectionInfo.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.TestRunnerConnectionInfo
+{
+    /// <summary>
+    /// Details of EndPoint address for connection b/w test runtime, test runnner
+    /// </summary>
+    public struct ConnectionInfo
+    {
+        /// <summary>
+        /// Endpoint where the service is hosted, This endpoint is specific to Transport Protocol
+        /// e.g. 127.0.0.0:8080 for socktes
+        /// </summary>
+        public string Endpoint
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the role played by Application during communication.
+        /// </summary>
+        public ConnectionRole Role
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the trasport protocol supported by test runtime
+        /// </summary>
+        public TransportChannel Channel
+        {
+            get;
+            set;
+        }
+    }
+
+    /// <summary>
+    /// Specifies the role of application, whether as host or client
+    /// </summary>
+    public enum ConnectionRole
+    {
+        /// <summary>
+        /// Act as server host
+        /// </summary>
+        Host = 0,
+
+        /// <summary>
+        /// Act as client
+        /// </summary>
+        Client = 1
+    }
+
+    /// <summary>
+    /// Specifies the underneath Transport channel to be used
+    /// E.g. For UWP/Desktop it is Sockets, for Android it could be SSH
+    /// </summary>
+    public enum TransportChannel
```

Comment:
Yes, since subsequent entries will follow below the enum, we are okay to change the enum in OM